aqt/archives.py

```diff
@@ -407,11 +405,11 @@ def _parse_update_xml(self, os_target_folder, update_xml_text, target_packages:
         for packageupdate in package_updates:
             if not self.all_extra:
                 target_packages.remove_module_for_package(packageupdate.name)
-            should_filter_archives: bool = self.subarchives and self.should_filter_archives(packageupdate.name)
+            should_filter_archives: bool = bool(self.subarchives) and self.should_filter_archives(packageupdate.name)
 
             for archive in packageupdate.downloadable_archives:
                 archive_name = archive.split("-", maxsplit=1)[0]
-                if should_filter_archives and archive_name not in self.subarchives:
+                if should_filter_archives and self.subarchives is not None and archive_name not in self.subarchives:
```

Comment:
oops, this was a merge error:
```suggestion
                if should_filter_archives and bool(self.subarchives) and archive_name not in self.subarchives:
```